### DISCUSSION TITLE
Feature backlog

### DISCUSSION THREAD

**Discussion by iappyx** [iappyx/Instrumenta#9] [Ideas]
These features are on the backlog:

**Backlog**
- Standard (light) graph formats

**Comment by jlcoomans** [comment]
HI Iappyx, first of all. thanks so much for building this tool. Truely a life saver in post-consulting jobs :)

Kind request for the backlog; I noticed that several standard commands are not working like they do on the home tab of PPT. 
Example; if I have text selected;
- On the home tab of PPT, if I use the "Bullets" button, the standard version of bullets would directly be applied to selected text
- On the instrumeta tab, using the "bullets" button results in a selection pane where one first has to select the desired bullets.

In other words, some buttons in instrumenta lack the functionality to directly apply the standard/last used format.

Other buttons on the instrumenta tab that I noticed with this behaviour are; 
- Font colour
- Bullets
- Numbering
- Shape Fill
- Shape outline
- Table shading

My personal preferrence would be the functionality of the home tab buttons to prevents some additional clicks. 
Thanks for considering!


![image](https://user-images.githubusercontent.com/83705634/123385333-b3e67380-d595-11eb-8b4a-faef3f1b63fe.png)

**Reply by iappyx** [reply]
Hi jlcoomans, thanks for your nice feedback! 

I encountered the same issues in the ribbon, but this behaviour was actually updated in Instrumenta a few weeks ago (June 6th update)

Judging from your screenshots, I have the feeling that you are not using the latest version. Can you check under Advanced and then the questionmark button? The version dialog should say **version 1.0.**

You can solve it by installing the latest version.

**Reply by jlcoomans** [reply]
Amazing! You beat me to it. I was indeed running the old version (as I was kinda postponing updating to the latest version as I didnt knew how). Updated now to 1.0 and all fixed.  Thanks!

**Reply by iappyx** [reply]
Great! 

Upgrading is basically replacing the add-in file when PPT is closed, or removing the old add-in via PPT and then adding the new one. Since the filename changed in the final release of v1.0 it might be a bit cumbersome. Will add some info to the readme.

**Comment by richooms** [comment]
Hi iappyx,

I really love your tool and use it a lot in my management consulting job. I would love it if comments could be converted to notes on all slides. So adding an extra option to have the lowest function of this screenshot in bulk. 

<img width="323" alt="image" src="https://user-images.githubusercontent.com/3234392/130810696-11dc60b7-eb95-46c9-b9fe-6506982653dc.png">

**Reply by iappyx** [reply]
Thanks! Should not be that difficult. I'll look into it.

**Reply by iappyx** [reply]
> Hi iappyx,
> 
> I really love your tool and use it a lot in my management consulting job. I would love it if comments could be converted to notes on all slides. So adding an extra option to have the lowest function of this screenshot in bulk.
> 
> <img alt="image" width="323" src="https://user-images.githubusercontent.com/3234392/130810696-11dc60b7-eb95-46c9-b9fe-6506982653dc.png">

This feature is now part of the latest beta version (1.01). See how you can install the latest beta [here](https://github.com/iappyx/Instrumenta#how-to-install). Will be part of the next major release.

**Comment by jhoedt** [comment]
Thanks for creating a great tool! As a pixel-perfect perfectionist, there's only one thing still slowing me down, and that's positioning. Could that be integrated in the tool as well?

![image](https://user-images.githubusercontent.com/27055899/140281160-102f6f64-44f4-4287-9b20-695c8db7f926.png)

**Reply by iappyx** [reply]
Thanks! I tried to look into it earlier, but I have not been able to identify the right CustomUI ribbon object for these. We might be able to create a custom button that opens the standard positioning pane. Will look into it.

**Reply by iappyx** [reply]
Unfortunately the built-in controls cannot be used. But, in the latest beta v1.28, you can now open a floating dialog which you can use to set the position of shapes.

![afbeelding](https://user-images.githubusercontent.com/22550917/216473688-29d26c00-6d11-493f-bc99-70f73b9f2b5e.png)

**Comment by richooms** [comment]
Hi iappyx, 
Would it be possible to make a button that sets the font on all slides in the same font? Would be really nice!
Thanks in advance!

**Reply by iappyx** [reply]
Hi @richooms, first of all, thank you very much for your support!

There is a function in Instrumenta that might help. It can be found in the "Text" section. It's actually a shortcut to a standard function in PowerPoint. See below screenshots. Is this what you need?

![screenshot-1](https://user-images.githubusercontent.com/22550917/158819241-8dc84a04-488e-4b70-ba72-03e325aacaf4.jpg)
![screenshot-2](https://user-images.githubusercontent.com/22550917/158819262-8a18a38d-198b-45aa-98b5-03d9f8dc4640.jpg)

**Reply by richooms** [reply]
Would be exactly what I need. But I don't see it in the Mac version of instrumenta (just reinstalled the last beta version to ensure I'm up to date) 
![image](https://user-images.githubusercontent.com/3234392/158998430-9355ec3a-6f68-4e40-8bea-39c7a46e733b.png)

**Reply by iappyx** [reply]
You are right. There seems to be a difference between Mac and Windows functionality of Powerpoint. In Powerpoint for Mac you can use Edit --> FInd --> Replace fonts. Does that do the trick? If yes, I can try to add a shortcut to that Powerpoint functionality to the Mac version of Instrumenta.

![afbeelding](https://user-images.githubusercontent.com/22550917/159017439-a3d28be0-77ef-44e0-88b5-991b870b6fe6.png)

**Reply by richooms** [reply]
It does! Would be great if you could do so :)

**Reply by iappyx** [reply]
Have been looking into it, but do not have a solution yet. Need to find the Mac solution for using the code below:

`Application.CommandBars.ExecuteMso ("FontsReplaceFonts")`

This code works on Windows but is sadly not supported on Mac.

**Comment by jlcoomans** [comment]
HI Iappyx,

Kind request. I often need to adjust shape or cell margins. The feature currently part of instrumenta to change this uses steps of 0,01cm. For me this stepsize is to small. As a result, I still use the format shape view and manually enter the new margins. For the Instrumenta feature to be of added value for me I would suggest changing the stepsize to 0,05 (or maybe 0,10)

![image](https://user-images.githubusercontent.com/83705634/167136526-12dbf9e0-b82f-45b6-bacb-fcc37b044621.png)

Thanks for considering.

**Reply by iappyx** [reply]
Hi @jlcoomans, in the latest release 1.2 there is now a settings dialog where you can change this behavior. Check it out ;-)

![afbeelding](https://user-images.githubusercontent.com/22550917/181375336-5bf5a4d8-032d-4b8d-a0e2-1429e542ea6a.png)

**Reply by iappyx** [reply]
Depending on your locale there still is a small bug in this code for these step sizes. You can only use full numbers (no digits) at this moment in time if your locale decimal point is a comma. Still usable, but will fix soon.

**Reply by iappyx** [reply]
Updated the feature a bit in the latest version [1.21](https://github.com/iappyx/Instrumenta/releases/tag/1.21) 

Step sizes are now correct in every locale.

**Comment by TobiasBenz** [comment]
Hey hey @iappyx ,

I really love your tool eventhough I am not yet using it at its full power. But one thing that might help others as well is the option in the remove section to have something like removeHiddenSlides. Currenly I do it with another VBA like that to create a client version to send out:

Sub deleteHiddenSlides(sFileName As String)

Dim oTempPres As Presentation
 ActivePresentation.SaveCopyAs sFileName

Set oTempPres = Presentations.Open(sFileName)

Dim x As Long
For x = oTempPres.Slides.Count To 1 Step -1
    With oTempPres.Slides(x)
       If .SlideShowTransition.Hidden = True Then
          .Delete
       End If
    End With
Next
    oTempPres.Save
    oTempPres.Close
End Sub

**Reply by iappyx** [reply]
Hi @TobiasBenz, thanks!

Good one, This should be fairly easy to implement. Will look into it.

**Reply by iappyx** [reply]
@TobiasBenz this feature is added to the latest version [1.21](https://github.com/iappyx/Instrumenta/releases/tag/1.21)

**Comment by Vadigor** [comment]
As someone dealing with multiple client languages, beefing up the Set Slide Language functionality would be very useful. Specifically a way to limit/restrict the drop-down would be interesting (not sure if I can modify this myself in an ppam file?). But also behaviour to adjust only selected slides might be interesting. (I can often end up with a mix of reference slides in English and project slides in the local language. 😉)

**Reply by iappyx** [reply]
Hi @Vadigor. Thanks for the feedback. Yes, as a quick fix, you can change the code yourself if you want 😀 In any case I will look into it. Think we can make it a setting in the new settings menu (e.g. favorite languages). Applying it only to a subset of slides should also be fairly easy to implement. Will look into it beginning of September.

**Comment by DeLaGardie** [comment]
Hi! Just started using Instrumenta and it seems great - it has almost all the features I am used to from management consulting. One major thing I am missing though is **keyboard shortcuts** (beyond standard alt+...), e.g. binding align left to ctrl+alt+left arrow. To me this is a key productivity unlock. Is this a feature that is actually there, just not me finding it, or is it being developed? If not, does anyone know a way to bind PPT functions to shortcut keys?

Thanks again for a great add-in!

**Reply by iappyx** [reply]
Hi @DeLaGardie, this is definitely something I would like to add (it's not there yet). I believe you can do it manually in PowerPoint need to find out if this can also be programmed in VBA. Would also need to know which keyboard shortcuts are out there that people like to use.

**Reply by DeLaGardie** [reply]
Cool! What do you mean by manually in PPT? Something I could do myself?

Btw, happy to share input on shortcuts we had my previus job (MBB) if youre interested?

**Reply by iappyx** [reply]
Do you have a list of shortcuts?

**Reply by iappyx** [reply]
I have been working on a keyboard shortcut companion called [Instrumenta Keys](https://github.com/iappyx/Instrumenta-Keys). It only works on Windows and is highly experimental.

**Comment by EBITDAFF** [comment]
Hi iappyx, 

First, thank you so much for Instrumenta, this is a real game changer and can’t work without it anymore!

For example, I really like to use the following features:
-	Swap shapes
-	Set to largest, set to smallest 
-	Set dictionary
-	Copy/paste position and size
-	Align images at center of tables (one of my favorites)
-	Add note
-	Remove text from shapes
-	Select by fill color
-	Remove vertical/horizontal gap

If I had to find some improvements I would suggest:
-	Restricting dictionary list
-	An equivalent of the format painter for tables’ fill color/borders (this would be really useful)
-	Automatically cropping png by removing empty spaces
-	Apply same crop to different images
-	Make shortcuts easier to use
-	Make a menu to set note style (border, fill color, font size)

Anyway, thank you again so much for this tool, I use it every day and really consider how lucky I am to use this amazing tool for free!

**Reply by iappyx** [reply]
Great that you enjoy using Instrumenta!

Will look into these improvements. Not sure if they are all possible (e.g. cropping png by empty space might not be possible with VBA; and also the format painter for tables would be difficult, but there might be options).

**Comment by andant513** [comment]
Hello iappyx, first of all thanks A LOT for creating this amazing tool. 

I was wondering if it would be possible to add a button to save a ppt deck where you can store multiple assets/ designs (a slide library if you will). Then this deck could be easily invoked with another button. 

I already keep a slide library but I have to save it on the desktop and it’s not a very elegant solution

**Reply by iappyx** [reply]
Thank you! 

Should be possible, will look into it.

**Reply by iappyx** [reply]
Now part of the latest beta v1.26

**Comment by jlcoomans** [comment]
Hi Iappyx,

Small request for the backlog. Would it be possible to show the emoticons under the Emoji dropdown menu in colour instead of black/white? 

Im using emoji's for status updates in text. Would be great to directly see which "Large Circle" is Green/Yellow/Red without having to hover over them.

Thank's in advance.

**Reply by iappyx** [reply]
I've tried this. At some point it worked in color, but then it changed in one of the newer versions of PowerPoint. I've tried to add the [variation selector](https://en.wikipedia.org/wiki/Variation_Selectors_(Unicode_block) ) to force displaying a color emoji, but that does not work...

**Comment by gimlei** [comment]
First of all - thank you SO MUCH for this tool - it is by far the best that I have seen, including free and paid, and I say that coming from a long time at MBB. The fact that this is a live project with new features being added regularly is absolutely amazing - thanks again!

A couple of feature requests:
1. Keyboard shortcuts - in line with what was asked before, is it possible to add customizable shortcuts for Instrumenta features?
2. Sanitization / anonymization - right now it's an all or nothing for the entire deck. Would it be possible to introduce this feature (1) for selected slides only, and (2) for selected objects on a slide only?
3. Add settings to customize placement and formatting of the steps counter on the pages
4. Add settings to customize size and color of Harvey balls
5. If there is a changelog that can be made available to easily check feature changes from version to version, that would be amazing as well.

Lastly, completely off topic, but if anyone is aware of a similar type of Excel add-in (feature-rich, open-source, actively updated) that has things like formula auditing, auto-formatting, etc., please let me know!

Thank you again @iappyx for everything you do!!

**Reply by iappyx** [reply]
Hi @gimlei you're welcome, happy that you like it!

With regard to your requests:
1.  This is unfortunately not possible. VBA does not allow this in PowerPoint. You could use 3rd party software or do something like this: #37. I would love to add it, but I would mean a total rewrite in some other language. A step that I am considering, but will take a significant amount of time.
2. That would be possible, will look into it.
3. What would you like to customize in terms of placement? They are now just added in the top right corner, so you can drag them anywhere. Do you have another use case / example? Formatting I can look into.
4. That should also be possible. They are shapes that you can easily recolor. If you already have harvey balls on the slide, you can click them and 'update' them by changing the harvey ball. They will retain their size.
5. True, I only do that for releases and only when I have time to do it :-) Still on my backlog to create some content on how to use Instrumenta.

**Reply by iappyx** [reply]
@gimlei, your second suggestion is now implemented in the latest beta (v1.41)!

**Reply by iappyx** [reply]
I have been working on a keyboard shortcut companion called [Instrumenta Keys](https://github.com/iappyx/Instrumenta-Keys). It only works on Windows and is highly experimental.
